ast/substitute/Substitute.cc

```diff
@@ -42,36 +42,41 @@ class SubstWalk {
     }
 
 public:
-    SubstWalk(const core::GlobalSubstitution &subst) : subst(subst) {}
+    SubstWalk(T &subst) : subst(subst) {}
 
     ExpressionPtr preTransformClassDef(core::MutableContext ctx, ExpressionPtr tree) {
-        auto *original = cast_tree<ClassDef>(tree);
-        original->name = substClassName(ctx, std::move(original->name));
-        for (auto &anc : original->ancestors) {
+        auto &original = cast_tree_nonnull<ClassDef>(tree);
```

Comment:
`cast_tree_nonnull` avoids a tag check at runtime (it's still checked in debug builds). We _know_ that this tree is a ClassDef.